homeassistant/components/fibaro/sensor.py

```diff
@@ -95,56 +96,34 @@ async def async_setup_entry(
 class FibaroSensor(FibaroDevice, SensorEntity):
     """Representation of a Fibaro Sensor."""
 
-    def __init__(self, fibaro_device):
+    def __init__(self, fibaro_device: Any) -> None:
         """Initialize the sensor."""
-        self.current_value = None
-        self.last_changed_time = None
         super().__init__(fibaro_device)
         self.entity_id = ENTITY_ID_FORMAT.format(self.ha_id)
+
+        self._attr_native_unit_of_measurement = None
         if fibaro_device.type in SENSOR_TYPES:
-            self._unit = SENSOR_TYPES[fibaro_device.type][1]
-            self._icon = SENSOR_TYPES[fibaro_device.type][2]
-            self._device_class = SENSOR_TYPES[fibaro_device.type][3]
+            self._attr_native_unit_of_measurement = SENSOR_TYPES[fibaro_device.type][1]
+            self._attr_icon = SENSOR_TYPES[fibaro_device.type][2]
+            self._attr_device_class = SENSOR_TYPES[fibaro_device.type][3]
             self._attr_state_class = SENSOR_TYPES[fibaro_device.type][4]
```

Comment:
It looks like `SensorEntityDescription` would be a nice cleanup here

Reply:
@bdraco Thanks for looking at my PR. I thought as well about `SensorEntityDescription`, but don't know how to use it here. I already know that they should be constant and be a tuple. So I can add a key to the description to find the correct one and resolve it. But what about unknown things, for example we always add the sensor just with the unit served from fibaro if we don't know the type. The user can then customize the device correctly in HA. Do you have a hint for me how this should be implemented?

Reply:
I think you can replace the list below with a custom `SensorEntityDescription`.  `august` and other integrations have examples of adding a mixin to augment with additional fields

```
    "com.fibaro.energyMeter": [
        "Energy",
        ENERGY_KILO_WATT_HOUR,
        None,
        SensorDeviceClass.ENERGY,
        SensorStateClass.TOTAL_INCREASING,
    ],
```

Reply:
Implemented with SensorEntityDescription.